### PR TITLE
Add/remove listing in/from elasticsearch index

### PR DESCRIPTION
**Pull request by rhnonose**
- Add listing to elasticsearch index when it's activated
- Remove listing from elasticsearch index when it's deactivated
- Put in a `GenServer` for async

In the future, we might use the same `GenServer` to listen to the mutation using subscriptions. There's no clear interface as of now in `absinthe` to make that trivial.

### CODE REVIEW

File: lib/re/application.ex [hunk 1]
```diff
@@ -22,7 +22,8 @@ defmodule Re.Application do
       supervisor(Absinthe.Subscription, [ReWeb.Endpoint]),
       worker(Visualizations, []),
       worker(Emails.Server, []),
-      Cluster
+      worker(Search.Server, []),
+      Search.Cluster
       # worker(Elasticsearch.Executable, [
```
Comment (gusaiani) [inline lib/re/application.ex]: @rhnonose Shouldn't we just delete these commented lines?

File: test/re/listings/listings_test.exs [hunk 2]
```diff
@@ -206,16 +206,16 @@ defmodule Re.ListingsTest do
       assert retrieved_listing.user_id == user.id
     end
 
-    test "should activate for admin user" do
+    test "should insert inactivate for admin user" do
```
Comment (gusaiani) [inline test/re/listings/listings_test.exs]: @rhnonose `inactivate`?

File: test/re/listings/listings_test.exs [hunk 3]
```diff
@@ -206,16 +206,16 @@ defmodule Re.ListingsTest do
...
       user = insert(:user, role: "admin")
 
       assert {:ok, inserted_listing} = Listings.insert(@insert_listing_params, address, user)
       assert retrieved_listing = Repo.get(Listing, inserted_listing.id)
-      assert retrieved_listing.is_active
+      refute retrieved_listing.is_active
     end
 
-    test "should not activate for normal user" do
+    test "should insert inactivate for normal user" do
```
Comment (gusaiani) [inline test/re/listings/listings_test.exs]: @rhnonose `inactivate`?

File: test/re/listings/listings_test.exs (hunk 2 above)
Comment (gusaiani) [reply test/re/listings/listings_test.exs]: should it be `deactivate`?

File: lib/re/application.ex (hunk 1 above)
Comment (rhnonose) [reply lib/re/application.ex]: It's good to have there if we want to execute elasticsearch/kibana together with the application like I do.

File: test/re/listings/listings_test.exs (hunk 2 above)
Comment (rhnonose) [reply test/re/listings/listings_test.exs]: No, this is for adding new listings.

File: test/re/listings/listings_test.exs (hunk 3 above)
Comment (rhnonose) [reply test/re/listings/listings_test.exs]: Same as above.

File: test/re/listings/listings_test.exs (hunk 2 above)
Comment (gusaiani) [reply test/re/listings/listings_test.exs]: @rhnonose `inactive` then?

File: test/re/listings/listings_test.exs [hunk 4]
```diff
@@ -182,6 +182,16 @@ defmodule Re.ListingsTest do
     end
   end
 
+
```
Comment (gusaiani) [inline test/re/listings/listings_test.exs]: There should be no more than 1 consecutive blank lines.

**Comment by gusaiani** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/em-casa/backend/pulls/188.

**Pull request merged**